program/src/instruction.rs

```diff
@@ -28,15 +28,22 @@ pub enum PaladinGovernanceInstruction {
     /// 0. `[s]` Validator vote account.
     /// 1. `[ ]` Paladin stake account.
     /// 2. `[w]` Proposal account.
+    // Jon: we'll definitely need a way to add instructions to a proposal while it's
+    // in "draft" phase. We can add it to this instruction, but then we're limited by
+    // the total transaction size when it comes to the proposal. For max flexibility,
+    // we can add a separate governance instruction to add an instruction to a proposal
+    // in the "draft" phase, signed by the author.
     CreateProposal,
+    // Jon: to go with the state machine concept, we'll need an instruction to move a
+    // proposal from "draft" to "voting", signed by the proposal author
     /// Cancel a governance proposal.
     ///
     /// Stake account provided must be the proposal creator.
     ///
     /// Accounts expected by this instruction:
     ///
     /// 0. `[s]` Validator vote account.
-    /// 1. `[ ]` Paladin stake account.
+    /// 1. `[ ]` Paladin stake account. // Jon: is this account needed?
```

Comment:
I don't know - the document stated a stake account was required for create and cancel, so I'm just validating it's an actual stake account and belongs to the authority (author) provided.

Reply:
Ah gotcha. It's not a huge deal, but since the stake account is required to create the proposal, and we store the creator on the proposal, we are implicitly checking that there's a stake account during cancel

Reply:
This account was removed.